checks/run_checks.go

```diff
@@ -40,6 +40,8 @@ func Run(ctx context.Context, client *kube.Client, checkFilter CheckFilter, diag
 	if len(all) == 0 {
 		return nil, errors.New("No checks to run. Are you sure that you provided the right names for groups and checks?")
 	}
+	objectFilter.Filter(objects)
```

Comment:
I'd rather let the apiserver do the filtering instead of filtering on the client side. We can do this by passing the object filter though to `kube.FetchObjects` and having it construct a [`fields.Selector`](https://godoc.org/k8s.io/apimachinery/pkg/fields#Selector) to use in the `ListOptions` we pass for namespace objects.

I like the `ObjectsFilter` type - it will let us implement other object filters in the future if desired.

Reply:
Thanks for a good suggestion!
I use ListOptions for namespace scoped objects.